src/decision/src/subroutines/Subroutine.cpp

```diff
@@ -6,22 +6,22 @@
 
 #include "Subroutine.h"
 
-Subroutine::Subroutine(int argc, char** argv, std::string node_name) {
-    ros::init(argc, argv, node_name);
-}
+Subroutine::Subroutine() {}
 
 void Subroutine::startup() {
     ros::NodeHandle nh;
+    ros::NodeHandle private_nh("~");
+
     setupSubscriptions(nh);
 
-    ros::NodeHandle private_nh("~");
-    std::string topic   = private_nh.resolveName("sub_control");
-    uint32_t queue_size = 1;
+    std::string topic   = private_nh.resolveName("output");
+    uint32_t queue_size = 10;
     publisher_ = private_nh.advertise<geometry_msgs::Twist>(topic, queue_size);
 }
 
 void Subroutine::shutdown() {
-    ros::shutdown();
+    publisher_.shutdown();
+    subscriber_.shutdown();
```

Comment:
Yeah having the generic subscriber is good her because it allows us to avoid implementing shutdown the exact same way in every implementer. That's a good point about the single subscriber and I thought about that, it's fairly easy to change this to be a vector and have a foreach loop for shutting them down. I didn't do it at first for this PR because our subroutines currently just subscribe to one thing but I will add it in.

Reply:
fixed